src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,222 @@
+                self.adaptive_std_module = self._get_adaptive_std_network()
+
+    def _get_shared_mean_std_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim * 2,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._init_shared_b,
+            layer_normalization=self._layer_normalization
+        )
+
+    def _init_shared_b(self, b):
+        nn.init.zeros_(b[..., :self._action_dim])
+        nn.init.constant_(b[..., self._action_dim:], self._init_std_param.item())
+
+    def _get_mean_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+    def _get_adaptive_std_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._std_hidden_sizes,
+            hidden_nonlinearity=self._std_hidden_nonlinearity,
+            hidden_w_init=self._std_hidden_w_init,
+            hidden_b_init=self._std_hidden_b_init,
+            output_nonlinearity=self._std_output_nonlinearity,
+            output_w_init=self._std_output_w_init,
+            output_b_init=self._init_adaptive_b,
+            layer_normalization=self._layer_normalization)
+
+    def _init_adaptive_b(self, b):
+        return nn.init.constant_(b, self._init_std_param.item())
+
+    def forward(self, inputs):
+        if self._std_share_network:
+            output = self.shared_mean_std_network(inputs)
+            mean = output[..., :self._action_dim]
+            std = output[..., self._action_dim:]
+        else:
+            mean = self.mean_module(inputs)
+
+            if self._adaptive_std:
+                std = self.adaptive_std_module(inputs)
+            else:
+                broadcast_shape = list(inputs.shape[:-1]) + [self._action_dim]
+                x = torch.zeros(*broadcast_shape)
+                std = x + self._init_std_param
+
+        if self._min_std_param or self._max_std_param:
+            std = std.clamp(min=to_scalar(self._min_std_param), max=to_scalar(self._max_std_param))
+
+        if self._std_parameterization == 'exp':
+            log_std_var = std
+        else:
+            log_std_var = std.exp().add(1.).log().log()
```

Comment:
What is this actually doing?
Please use `elif` to be more specific then raise `NotImplementedError` for all other cases.

Reply:
I think we already check the std_parameterization at __init__
and also followed tensorflow [GaussianMLPModule](https://github.com/rlworkgroup/garage/blob/3bf0b2b40c3db30c66f95460f0280b442457cc7c/src/garage/tf/models/gaussian_mlp_model.py#L219).
If you think it needs to be fixed, I will change the code

Reply:
Ok thats fine